Set environment vars to set vendor-specific dandi-schema used by the DANDI instance

candleindark opened the pull request:
This PR is comparable to https://github.com/dandi/dandi-infrastructure/pull/224 in https://github.com/dandi/dandi-infrastructure. See commit message for details.

TODOs:

- [x] `DANDI_INSTANCE_NAME` env var is set to `"LINC"` for both production and staging. Make sure that it is the proper value
    - [x] The value of `"LINC-STAGING"` is used for the staging instance.
- ~~[ ] `DANDI_DOI_PREFIX` env var is set to `"10.80507"` in staging. Make sure that it is the proper value.~~
- [x] dandi-archive being able to set the config from DJANGO_DANDI_ variables (PR: TODO)
- [x] Add environment variable needed for vendorization of the publisher for `to_datacite`, https://github.com/dandi/dandi-schema/issues/303

Note: all DOI related environment vars have been removed. See individual commit messages for details.

candleindark commented on terraform/api.tf:
```diff
@@ -25,6 +25,7 @@ module "api" {
     django_cors_origin_regex_whitelist = ["^https:\\/\\/[0-9a-z\\-]+\\.netlify\\.app$"]
 
     additional_django_vars = {
+        DANDI_INSTANCE_NAME                            = "LINC"
```
Is `"LINC"` the proper value?

kabilar replied:
Yes, I believe so.  We use that instance name in the DANDI CLI: https://github.com/dandi/dandi-cli/blob/03b7c0720329b559431b0872537ded0fc2c13dc9/dandi/consts.py#L135-L139

candleindark commented on terraform/staging_pipeline.tf:
```diff
@@ -24,6 +24,8 @@ module "api_staging" {
   django_cors_origin_regex_whitelist = ["https://staging--gui-staging-lincbrain-org.netlify.app"]
 
   additional_django_vars = {
+    DANDI_INSTANCE_NAME                            = "LINC"
```
Is `"LINC"` the proper value?

candleindark replied:
value changed to `"LINC-STAGING"`.

candleindark commented on terraform/staging_pipeline.tf:
```diff
@@ -24,6 +24,8 @@ module "api_staging" {
 
   additional_django_vars = {
+    DANDI_INSTANCE_NAME                            = "LINC"
+    DANDI_DOI_PREFIX                               = "10.80507"
```
Does `DANDI_DOI_PREFIX` need to be set in staging? The reason I set it in staging and not setting it in production (the `api.tf`) is because `DJANGO_DANDI_DOI_API_PREFIX` has a valid value in staging.

candleindark replied:
Setting `DANDI_DOI_PREFIX` here is no needed. `DJANGO_DANDI_DOI_API_PREFIX` actually is not needed here either.

candleindark commented:
@kabilar @yarikoptic Please take a look at this PR.

candleindark committed:
feat: remove all env vars related to DOI settings in production

These vars, with the exception of `DJANGO_DANDI_DOI_PUBLISH`, need to be provided
together or not at all per
https://github.com/dandi/dandi-archive/blob/b7288ec920b0c2cf2efd9c6d43e4edbf5016885d/dandiapi/api/checks.py#L10-L25. Since LINC
doesn't publish Dandiset with DOI, it is
reasonable not to set these env var.
Additionally, the value for
`DJANGO_DANDI_DOI_API_PREFIX` before this
commit doesn't pass the pattern requirement
imposed for DOI prefix being introduced in
https://github.com/dandi/dandi-schema/pull/294.

candleindark committed:
feat: remove all env vars related to DOI settings in staging

These vars, with the exception of `DJANGO_DANDI_DOI_PUBLISH`, need to be provided
together or not at all per
https://github.com/dandi/dandi-archive/blob/b7288ec920b0c2cf2efd9c6d43e4edbf5016885d/dandiapi/api/checks.py#L10-L25. Since LINC
doesn't publish Dandiset with DOI, it is
reasonable not to set these env var.
Additionally, the value for
`DJANGO_DANDI_DOI_API_PREFIX` before this
commit doesn't pass the pattern requirement
imposed for DOI prefix being introduced in
https://github.com/dandi/dandi-schema/pull/294.

candleindark committed:
feat: set env vars to set vendor-specific dandi-schema in production

Set env vars need to set the vendor-specific dandi-schema
introduced by https://github.com/dandi/dandi-schema/pull/294

candleindark committed:
feat: set env vars to set vendor-specific dandi-schema in staging

Set env vars need to set the vendor-specific dandi-schema
introduced by https://github.com/dandi/dandi-schema/pull/294

yarikoptic commented on terraform/api.tf:
```diff
@@ -36,13 +36,10 @@ module "api" {
-        DJANGO_DANDI_DOI_API_URL                       = "https://api.datacite.org/dois"
-        DJANGO_DANDI_DOI_API_USER                      = "temp.dandi"
-        DJANGO_DANDI_DOI_API_PREFIX                    = "temp"
-        DJANGO_DANDI_DOI_PUBLISH                       = "true"
```
@kabilar would that be ok?

kabilar replied:
Thanks yea that works, since we don't generate DOIs.

kabilar commented on terraform/api.tf:
```diff
@@ -36,13 +36,10 @@ module "api" {
         DJANGO_SENTRY_DSN                              = "https://833c159dc622528b21b4ce4adef6dbf8@o4506237212033024.ingest.sentry.io/4506237213212672"
         DJANGO_SENTRY_ENVIRONMENT                      = "production"
         DJANGO_CELERY_WORKER_CONCURRENCY               = "4"
+        DJANGO_DANDI_INSTANCE_NAME                     = "DANDI"
```
```suggestion
        DJANGO_DANDI_INSTANCE_NAME                     = "LINC"
```

kabilar commented on terraform/staging_pipeline.tf:
```diff
@@ -35,13 +35,10 @@ module "api_staging" {
     DJANGO_SENTRY_DSN                              = "https://833c159dc622528b21b4ce4adef6dbf8@o4506237212033024.ingest.sentry.io/4506237213212672"
     DJANGO_SENTRY_ENVIRONMENT                      = "staging"
     DJANGO_CELERY_WORKER_CONCURRENCY               = "2"
+    DJANGO_DANDI_INSTANCE_NAME                     = "DANDI-STAGING"
```
```suggestion
    DJANGO_DANDI_INSTANCE_NAME                     = "LINC-STAGING"
```

candleindark committed:
Correct instance name to "LINC"

Co-authored-by: Kabilar Gunalan <kabilar.gunalan@gmail.com>

candleindark committed:
Correct instance name to "LINC-STAGING" for staging instance

Co-authored-by: Kabilar Gunalan <kabilar.gunalan@gmail.com>
